### PR TITLE
Feat/ml pipeline

### PR DESCRIPTION
**Pull request by aladin6060**
This is a first implementation of an image labeling pipeline with the ML kit.  So far the predictions are really bad but I am not sure if it's the model's fault or something with the image preprocessing goes wrong (i.e bad input to the model).

image

### CODE REVIEW

**Review by adam-narozniak — APPROVED** [review]
The code looks well structured. 
I'm not familiar with the ML details of the implementation, so I can't speak to that.

File: docs/ML_PIPELINE.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+A problem with both datasets is, that the hands are recorded infront a white surface or a green screen. Therfore the background of the pictures taken with the camera should be remove. This could be done with image segmentation but makes the whole pipline much more complicated.
+
+
+### 2. Pose estimation
+
+A second approach would be to get the pose of the hand and deduce if a rock, paper or scissor is shown. Media Pipe provoides such a [model](https://google.github.io/mediapipe/solutions/hands), specificly made for mobile usage. The challange for this approach is to map the pose to the three possible symbols. This could either be done with a second NN or hard coded.
+
+## Implementation
+
+Most of the implementation has been adapted from the the android [example app](https://github.com/googlesamples/mlkit/tree/master/android/vision-quickstart). Parts which were not used have been removed.
```
[the remark this answers is not in the document]
Comment (aladin6060) [reply docs/ML_PIPELINE.md:29]: Yes, I implemented the first option. I will modify it.

**Review by gaetschwartz — COMMENTED** [review]
Similarly to adam, I'm can't speak for the ML part, but nothing shocking here. LGTM. As for testing, do you have still some part left, or is everything you planned to test, tested ?

**Review by leonardopennino — APPROVED** [review]
The code looks well structured but I'm not an expert on ML so I cannot say anything more.

**Comment by panadax** [comment]
https://user-images.githubusercontent.com/33196494/170059808-256cc0e3-31d4-47ed-bbfa-9fff9af0919e.MOV

Great work! But it seems the precision is not so good at my machine..

**Pull request merged**